compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2888,11 +2888,11 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
       def Transparent: Flags = dotc.core.Flags.Transparent
 
       // Keep: aligned with Quotes's `newMethod` doc
-      private[QuotesImpl] def validMethodFlags: Flags = Private | Protected | Override | Deferred | Final | Method | Implicit | Given | Local | AbsOverride | JavaStatic // Flags that could be allowed: Synthetic | ExtensionMethod | Exported | Erased | Infix | Invisible
+      private[QuotesImpl] def validMethodFlags: Flags = Private | Protected | Override | Deferred | Final | Method | Implicit | Given | Local | AbsOverride | Synthetic | Artifact | ExtensionMethod | Exported | Erased | Infix | Invisible | JavaStatic
```

Comment:
```suggestion
      private[QuotesImpl] def validMethodFlags: Flags = Private | Protected | Override | Deferred | Final | Method | Implicit | Given | Local | AbsOverride | JavaStatic | Synthetic | Artifact // Flags that could be allowed: Synthetic | ExtensionMethod | Exported | Erased | Infix | Invisible
```
See other comment

Reply:
Does the comment apply to flags like `ExtensionMethod` or `Infix`? It seems to me that (at least) these flags can appear in code that is being expanded by the macro. So the macro has to deal with the flags being there, and the compiler phases down the pipeline have, too.

Similar for `Erased` (under `experimental.erasedDefinitions`): Adding a method with the `Erased` flag in a macro and trying to invoke the method can also trigger the expected "method is declared as erased, but is in fact used" error.

In what sense does the compiler not handle them?